FreeRTOS-Plus/Demo/Device_Shadow_Windows_Simulator/Device_Shadow_Demo/DemoTasks/shadow_demo_helpers.c

```diff
@@ -0,0 +1,932 @@
+                        eMqttStatus ) );
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    /* Close the network connection.  */
+    TLS_FreeRTOS_Disconnect( pxNetworkContext );
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+int32_t lSubscribeToTopic( const char * pcTopicFilter,
+                          uint16_t usTopicFilterLength )
+{
+    int returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t eMqttStatus;
+    MQTTContext_t * pxMqttContext = &mqttContext;
+    MQTTSubscribeInfo_t pSubscriptionList[ mqttexampleTOPIC_COUNT ];
+
+    assert( pxMqttContext != NULL );
+    assert( pcTopicFilter != NULL );
+    assert( usTopicFilterLength > 0 );
+
+    /* Start with everything at 0. */
+    ( void ) memset( ( void * ) pSubscriptionList, 0x00, sizeof( pSubscriptionList ) );
+
+    /* This example subscribes to only one topic and uses QOS1. */
+    pSubscriptionList[ 0 ].qos = MQTTQoS1;
+    pSubscriptionList[ 0 ].pTopicFilter = pcTopicFilter;
+    pSubscriptionList[ 0 ].topicFilterLength = usTopicFilterLength;
+
+    /* Generate packet identifier for the SUBSCRIBE packet. */
+    globalSubscribePacketIdentifier = MQTT_GetPacketId( pxMqttContext );
+
+    /* Send SUBSCRIBE packet. */
+    eMqttStatus = MQTT_Subscribe( pxMqttContext,
+                                  pSubscriptionList,
+                                  sizeof( pSubscriptionList ) / sizeof( MQTTSubscribeInfo_t ),
+                                  globalSubscribePacketIdentifier );
+
+    if( eMqttStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send SUBSCRIBE packet to broker with error = %u.",
+                    eMqttStatus ) );
+        returnStatus = EXIT_FAILURE;
+    }
+    else
+    {
+        LogInfo( ( "SUBSCRIBE topic %.*s to broker.\n\n",
+                   usTopicFilterLength,
+                   pcTopicFilter ) );
+
+        /* Process incoming packet from the broker. Acknowledgment for subscription
+         * ( SUBACK ) will be received here. However after sending the subscribe, the
+         * client may receive a publish before it receives a subscribe ack. Since this
+         * demo is subscribing to the topic to which no one is publishing, probability
+         * of receiving publish message before subscribe ack is zero; but application
+         * must be ready to receive any packet. This demo uses MQTT_ProcessLoop to
+         * receive packet from network. */
+        eMqttStatus = MQTT_ProcessLoop( pxMqttContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
+
+        if( eMqttStatus != MQTTSuccess )
+        {
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "MQTT_ProcessLoop returned with status = %u.",
+                        eMqttStatus ) );
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+int32_t lUnsubscribeFromTopic( const char * pcTopicFilter,
```

Comment:
All of these functions should start with `prv`

Reply:
I thought only static or private function with ```prv```.
[Reference](https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html): 
- Functions
File scope static (private) functions are prefixed with prv.
API functions are prefixed with their return type, as per the convention defined for variables, with the addition of the prefix v for void.
API function names start with the name of the file in which they are defined. For example vTaskDelete is defined in tasks.c, and has a void return type.
Is it correct?

Reply:
Yes, I thought this was part of the other file.